app/common/filters.py

```diff
@@ -0,0 +1,39 @@
+from datetime import date, datetime
+
+
+def format_date(value: date | datetime) -> str:
+    """Format a date or datetime as follows:
+
+    > Friday 16 May 2025
+    """
+    return value.strftime("%A %-d %B %-Y")
+
+
+def format_datetime(value: datetime) -> str:
+    """Format a datetime as follows:
+
+    > 10:37am on Friday 16 May 2025
+
+    If the datetime is exactly on the hour, then minutes will not be included:
+
+    > 10am on Friday 16 May 2025
+    """
+    fmt = "%-I:%M%p on %A %-d %B %-Y"
+
+    if value.minute == 0:
+        fmt = "%-I%p on %A %-d %B %-Y"
+
+    formatted_datetime = value.strftime(fmt)
+    formatted_datetime = formatted_datetime.replace("AM", "am").replace("PM", "pm")
+
+    return formatted_datetime
+
+
+def format_date_range(start: date, end: date) -> str:
+    from_, to = format_date(start), format_date(end)
+    return f"{from_} to {to}"
+
+
+def format_datetime_range(start: datetime, end: datetime) -> str:
+    from_, to = format_datetime(start), format_datetime(end)
```

Comment:
And a docstring here too please :)

Reply:
Yep yep, fair shout 👍  I don't know why I didn't - I think just because it was mostly handing off to the other functions in a fairly trivial way, but still good to have an example of the format I think